### PR TITLE
Add Core functions for semtype

### LINKED ISSUE #32538: Implement missing functions in Core class
**Description:**
Port missing functions in semtype Core class from core.bal

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #32538 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by ushirask** [comment]
Converted to draft since there is a possible bug in nballerina related to `simpleArrayMemberType` and `simpleMapMemberType`

File: semtypes/src/main/java/io/ballerina/semtype/Core.java [hunk 1]
```diff
@@ -382,7 +513,7 @@
         }
     }
 
-    public static boolean containsConstInt(SemType t, int n) {
+    public static boolean containsConstInt(SemType t, Long n) {
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/Core.java:516]: Shall we make this a `long`

File: semtypes/src/main/java/io/ballerina/semtype/Core.java [hunk 2]
```diff
@@ -298,34 +300,154 @@
...
     }
 
     public static Optional<Value> singleShape(SemType t) {
+        if (t == PredefinedType.NIL) {
+            return Optional.of(Value.from(null));
+        } else if (t instanceof UniformTypeBitSet) {
+            return Optional.empty();
+        } else if (isSubtypeSimple(t, PredefinedType.INT)) {
+            SubtypeData sd = getComplexSubtypeData((ComplexSemType) t, UniformTypeCode.UT_INT);
+             Optional<Long> value = IntSubtype.intSubtypeSingleValue(sd);
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/Core.java:434]: extra space

File: semtypes/src/main/java/io/ballerina/semtype/Core.java [hunk 3]
```diff
@@ -298,34 +300,154 @@
...
+        } else if (isSubtypeSimple(t, PredefinedType.STRING)) {
+            SubtypeData sd = getComplexSubtypeData((ComplexSemType) t, UniformTypeCode.UT_STRING);
+            Optional<String> value = StringSubtype.stringSubtypeSingleValues(sd);
+            return value.isEmpty() ? Optional.empty() : Optional.of(Value.from(value));
+        } else if (isSubtypeSimple(t, PredefinedType.BOOLEAN)) {
+            SubtypeData sd = getComplexSubtypeData((ComplexSemType) t, UniformTypeCode.UT_BOOLEAN);
+            Optional<Boolean> value = BooleanSubtype.booleanSubtypeSingleValue(sd);
+            return value.isEmpty() ? Optional.empty() : Optional.of(Value.from(value));
+        }
+        return Optional.empty();
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/Core.java:450]: can remove this line too, I guess

File: semtypes/src/main/java/io/ballerina/semtype/Core.java [hunk 4]
```diff
@@ -298,34 +300,154 @@
...
+            return Optional.of(Value.from(null));
+        } else if (t instanceof UniformTypeBitSet) {
+            return Optional.empty();
+        } else if (isSubtypeSimple(t, PredefinedType.INT)) {
+            SubtypeData sd = getComplexSubtypeData((ComplexSemType) t, UniformTypeCode.UT_INT);
+             Optional<Long> value = IntSubtype.intSubtypeSingleValue(sd);
+            return value.isEmpty() ? Optional.empty() : Optional.of(Value.from(value));
+        } else if (isSubtypeSimple(t, PredefinedType.FLOAT)) {
+            SubtypeData sd = getComplexSubtypeData((ComplexSemType) t, UniformTypeCode.UT_FLOAT);
+             Optional<Double> value = FloatSubtype.floatSubtypeSingleValue(sd);
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/Core.java:438]: extra space

File: semtypes/src/main/java/io/ballerina/semtype/subtypedata/StringSubtype.java [hunk 5]
```diff
@@ -38,15 +43,21 @@
         this.values = values;
     }
 
+    private StringSubtype(boolean allowed, EnumerableString value) {
+        this.allowed = allowed;
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/subtypedata/StringSubtype.java:47]: `this(allowed, new EnumerableString[]{value})`

File: semtypes/src/main/java/io/ballerina/semtype/subtypedata/FloatSubtype.java [hunk 6]
```diff
@@ -40,19 +40,20 @@
     public FloatSubtype(boolean allowed, EnumerableFloat value) {
         this.allowed = allowed;
         this.values = new EnumerableFloat[1];
-        values[0] = value;
+        this.values[0] = value;
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/subtypedata/FloatSubtype.java:43]: Shall we reuse the other constructor here

File: semtypes/src/main/java/io/ballerina/semtype/subtypedata/FloatSubtype.java [hunk 7]
```diff
@@ -40,19 +40,20 @@
...
     }
 
     public FloatSubtype(boolean allowed, EnumerableFloat[] values) {
         this.allowed = allowed;
         this.values = values;
     }
 
-    public static SemType floatConst(EnumerableFloat value) {
-        return PredefinedType.uniformSubtype(UniformTypeCode.UT_FLOAT, new FloatSubtype(true, value));
+    public static SemType floatConst(Double value) {
```
Comment (rdhananjaya) [inline semtypes/src/main/java/io/ballerina/semtype/subtypedata/FloatSubtype.java:51]: shall we use `double` as line:53 going to unbox this anyway.

**Review by rdhananjaya — APPROVED** [review]

**Pull request merged**